File: 每日一题/4.28/2302.统计得分小于-k-的子数组数目.py
#
# @lc app=leetcode.cn id=2302 lang=python3
#
# [2302] 统计得分小于 K 的子数组数目
#
from typing import List
# @lc code=start
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        i = 0
        res = 0
        sums = 0
        for j in range(len(nums)) :
            sums += nums[j]
            while sums * (j - i + 1) >= k :
                sums -= nums[i]
                i += 1
            res += j - i + 1
        return res
# @lc code=end

File: 每日一题/2025-4/4.28/2302.统计得分小于-k-的子数组数目.cpp
/*
 * @lc app=leetcode.cn id=2302 lang=cpp
 *
 * [2302] 统计得分小于 K 的子数组数目
 */
#include <bits/stdc++.h>
#include <ranges>
using namespace std;
// @lc code=start
class Solution {
public:
    long long countSubarrays(vector<int>& nums, long long k) {
        long long res = 0;
        long long sums = 0;
        int i = 0;
        for (int j = 0; j < nums.size(); j ++ )
        {
            sums += nums[j];
            while (sums * (j - i + 1) >= k)
            {
                sums -= nums[i ++];
            }
            res += j - i + 1;
        }
        return res;
    }
};
// @lc code=end

